orttraining/orttraining/test/gradient/gradient_op_test_utils.cc

```diff
@@ -202,18 +202,23 @@ void GradientOpTester::Run(
           const KernelCreateInfo* kci;
           auto st = reg->TryFindKernel(node, execution_provider->Type(), &kci);
           if (!st.IsOK()) {
-            auto* node_func = node.GetFunctionBody();
-            if (!node_func) {
+            if (!node.CanBeInlined()) {
               valid = false;
             } else {
-              for (auto& sub_node : node_func->Body().Nodes()) {
-                if (sub_node.OpType() != "Constant") {
-                  auto sub_reg = execution_provider->GetKernelRegistry();
-                  const KernelCreateInfo* sub_kci;
-                  st = sub_reg->TryFindKernel(sub_node, execution_provider->Type(), &sub_kci);
-                  if (!st.IsOK()) {
-                    valid = false;
-                    break;
+              std::unique_ptr<Function> node_func;
+              st = node.GetInstantiateFunctionBody(node_func);
+              if (!st.IsOK()) {
+                valid = false;
+              } else {
+                for (auto& sub_node : node_func->Body().Nodes()) {
```

Comment:
It looks like this piece of code doesn't handle nested function-calls? It should probably be defined as a recursive function to handles nested calls. I realize this is a limitation of the pre-existing code, unrelated to this PR refactoring. Maybe we can add a TODO comment here and track it separately for a different PR later.

Reply:
added.